components/project-card/project-card.jsx

```diff
@@ -41,33 +43,80 @@ class ProjectCard extends React.Component {
   }
 
   setInitialBookmarkedStatus() {
-    let bookmarks = getBookmarks();
+    if (user.loggedin) {
+      this.setState({
+        bookmarked: this.props.isBookmarked
+      });
+
+      return;
+    }
+
+    let bookmarks = bookmarkManager.bookmarks.get();
     let bookmarked;
-    let index;
 
     if (bookmarks) {
-      index = bookmarks.indexOf(this.props.id);
-      if (index > -1) {
-        bookmarked = true;
-      } else {
-        bookmarked = false;
-      }
+      bookmarked = bookmarks.indexOf(this.props.id) > -1;
       this.setState({bookmarked: bookmarked});
     }
   }
 
-  bookmarkProject(bookmarks) {
-    ReactGA.event(this.createGaEventConfig(`Bookmark button`, `Bookmarked`));
+  bookmarkProjectOnLs(bookmarks) {
     bookmarks.unshift(this.props.id);
     this.setState({bookmarked: true});
   }
 
-  unbookmarkProject(bookmarks,index) {
-    ReactGA.event(this.createGaEventConfig(`Bookmark button`, `Unbookmarked`));
-    bookmarks.splice(index,1);
+  unbookmarkProjectOnLs(bookmarks) {
+    bookmarks.splice(bookmarks.indexOf(this.props.id), 1);
     this.setState({bookmarked: false});
   }
 
+  toggleBookmarkedStateOnLs() {
```

Comment:
this is a bad function name, too, let's give that a more immediately obvious name.